tests/application_development/code_relocation/src/test_file1.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/sys/printk.h>
+#include <ztest.h>
+
+uint32_t var_sram2_data = 10U;
+uint32_t var_sram2_bss;
+K_SEM_DEFINE(test, 0, 1);
+const uint32_t var_sram2_rodata = 100U;
+
+__in_section(custom_section, static, var) uint32_t var_custom_data = 1U;
+
+extern void function_in_sram(int32_t value);
+void function_in_custom_section(void);
+
+ZTEST(code_relocation, test_function_in_sram2)
+{
+	/* Print values from sram2 */
+	printk("Address of var_sram2_data %p\n", &var_sram2_data);
+	printk("Address of k_sem_give %p\n", &k_sem_give);
+	printk("Address of var_sram2_rodata %p\n", &var_sram2_rodata);
+	printk("Address of var_sram2_bss %p\n\n", &var_sram2_bss);
```

Comment:
This is a test, so having a `printk` without checking anything is really not that useful. You should check here that for example `var_sram2_bss` is actually in the BSS, `var_sram2_rodata` in RODATA, etc...

Reply:
right, le tme check